src/metabase/lib/dispatch.cljc

```diff
@@ -0,0 +1,38 @@
+(ns metabase.lib.dispatch)
+
+(defn- mbql-clause-type [x]
+  (when (and (vector? x)
+             (keyword? (first x)))
+    (keyword (first x))))
+
+(defn- cljc-friendly-type-keyword
```

Comment:
I do realize this is somewhat duplicated with the Cljs-only version in `metabase.test-runner.assert-exprs.approximately-equal` but I thought I'd solicit feedback on what the "right" way to do this is before consolidating too much.

Reply:
Yeah, `(isa? "X" js/String)` is falsey so actually having the result of `(type  x)` doesn't work in Cljs

Reply:
`(isa? "X" String)` is false in Clojure. Why would that prevent us from dispatching on `(type _)`?
There are differences like
```
cljs.user=> (instance? js/String "X")
false
```
and
```
user=> (instance? String "X")
true
```
But does that mean that dispatching on type cannot work?

Reply:
I did run into some issues with this, but I guess maybe I wrote the wrong thing down here.

I think the root problem is that `(isa? class superclass)` is true in Clojure, as is `(isa? class interface)`, but not ClojureScript. A type (constructor in Cljs) is only `isa?` to itself. So AFAIK there's no way to do a generic dispatch for all map types the way we can with Clojure. In Clojure we can do `IPersistentMap` or whatever, but in Cljs that doesn't work, you'd have to have a separate method for `cljs.core/PersistentArrayMap` and `cljs.core/PersistentHashMap`, and any other map types that might exist

Reply:
You also can't have a generic/fallback method for `js/Object` like you can with `java.lang.Object`